### PR TITLE
Feature/superuser

### PR DESCRIPTION
**Pull request by annashamray**

### CODE REVIEW

File: docs/admin/authorization.rst [hunk 1]
```diff
@@ -88,3 +88,19 @@ fields you can submit the form.
...
 If you want to know how to use Objects API you can follow :ref:`api_usage`
+
+
+Superuser permissions
+----------------------
+
+It's possible to set superuser permissions in Objects API. A client with such permissions
+is able to request objects for all objecttypes.
+
+In the admin page of the Objects API go to the Token authorizations" resource and click on
```
Comment (Viicos) [inline docs/admin/authorization.rst:99]:
```suggestion
In the admin page of the Objects API go to the "Token authorizations" resource and click on
```

File: src/objects/api/fields.py [hunk 2]
```diff
@@ -42,7 +44,23 @@ def to_internal_value(self, data):
         try:
             return self.get_queryset().get_by_url(data)
         except ObjectDoesNotExist:
-            self.fail("does_not_exist", value=smart_text(data))
+            # if service is configured, but objec_type is missing
```
Comment (Viicos) [inline src/objects/api/fields.py:47]:
```suggestion
            # if service is configured, but object_type is missing
```

File: src/objects/api/fields.py [hunk 3]
```diff
@@ -42,7 +44,23 @@ def to_internal_value(self, data):
...
+            # let's try to create an ObjectType
+            service = Service.get_service(data)
+            if not service:
+                self.fail("does_not_exist", value=smart_text(data))
```
Comment (Viicos) [inline src/objects/api/fields.py:51]:
```suggestion
                self.fail("does_not_exist", value=smart_str(data))
```

I think when rebasing on the 4.2 upgrade the import will be updated already

File: docs/admin/authorization.rst [hunk 4]
```diff
@@ -88,3 +88,19 @@ fields you can submit the form.
 Now the client who has this token can access the objects with the "Boom" object type.
 
 If you want to know how to use Objects API you can follow :ref:`api_usage`
+
+
+Superuser permissions
+----------------------
+
+It's possible to set superuser permissions in Objects API. A client with such permissions
```
Comment (Viicos) [inline docs/admin/authorization.rst:96]:
```suggestion
It's possible to set up superuser permissions in Objects API. A client with such permissions
```

File: docs/admin/authorization.rst (hunk 1 above)
Comment (Viicos) [reply docs/admin/authorization.rst:99]: seems to be not applied

File: src/objects/api/fields.py [hunk 5]
```diff
@@ -42,7 +44,23 @@ def to_internal_value(self, data):
...
+            if not service:
+                self.fail("does_not_exist", value=smart_str(data))
+
+            uuid = get_uuid_from_path(data)
+            object_type = self.get_queryset().model(service=service, uuid=uuid)
+
+            try:
+                object_type.clean()
+            except ValidationError:
+                self.fail("does_not_exist", value=smart_str(data))
```
Comment (Viicos) [inline src/objects/api/fields.py:59]: Is the error code a copy paste mistake?

**Review by SonnyBA — APPROVED** [review]

File: src/objects/api/fields.py (hunk 5 above)
Comment (annashamray) [reply src/objects/api/fields.py:59]: No. In `object_type.clean()` we try to request the object type url. If it's not existed `ValidationError` will be raised.

**Review by Viicos — APPROVED** [review]

**Pull request merged**